flow/project.py

```diff
@@ -1692,18 +1692,88 @@ def _store_bundled(self, operations):
         else:
             h = '.'.join(op.id for op in operations)
             bid = '{}/bundle/{}'.format(self, sha1(h.encode('utf-8')).hexdigest())
-            fn_bundle = self._fn_bundle(bid)
+            fn_bundle = self._fn_stored('.bundles', bid)
             os.makedirs(os.path.dirname(fn_bundle), exist_ok=True)
             with open(fn_bundle, 'w') as file:
                 for operation in operations:
                     file.write(operation.id + '\n')
             return bid
 
+    def _store_aggregates(self, operations):
+        """Store aggregate-ids per operation information.
+
+        This enables status check of aggregates which were
+        formed previously but are not present currently for any
+        operation.
+
+        :param operations:
+            The operations to be submitted.
+        :type operations:
+            A sequence of instances of :py:class:`.JobOperation`
+        """
+        for operation in operations:
+            op_fn = '{}.txt'.format(operation.name)
+            aggregate_wid = '{} {} \n'.format(operation.id, ' '.join(map(str, operation._jobs)))
+            fn_aggregate = self._fn_stored('.aggregates', op_fn)
+            try:
+                os.makedirs(os.path.dirname(fn_aggregate), exist_ok=True)
+            except PermissionError:
+                continue
+            if os.path.exists(fn_aggregate):
+                with open(fn_aggregate, 'r+') as file:
+                    if aggregate_wid in file:
+                        continue
+                    else:
+                        file.write(aggregate_wid)
```

Comment:
You need a read call to store the file in a string first. Otherwise we have no guarantee that we will be at the end of the file when writing.
```python
agg_file_contents = file.read()
if aggregate_wid in agg_file_contents:
    continue
else:
    file.write(aggregate_wid)
````